File: Python/Samples/Visitor/ObjectStructure.py
import Elements
import Visitors


class ObjectStructure:
    def __init__(self):
        self.elements = []

    def attach(self,element: Elements.Element):
        self.elements.append(element)

    def detach(self,element: Elements.Element):
        self.elements.remove(element)

    def accept(self,visitor: Visitors.Visitor):
        for elm in self.elements:
            elm.accept(visitor)

File: CSharp/DP.Domain/Samples/Visitor/ObjectStructure.cs
using System.Collections.Generic;

namespace DP.Domain.Samples.Visitor
{
    public class ObjectStructure : IObjectStructure
    {
        public List<IElement> Elements { get; set; }

        public ObjectStructure()
        {
            this.Elements = new List<IElement>();
        }
        public void Attach(IElement element)
        {
            this.Elements.Add(element);
        }
        public void Detach(IElement element)
        {
            this.Elements.Remove(element);
        }
        public void Clear()
        {
            this.Elements.Clear();
        }
        public void Accept(Visitor visitor)
        {
            this.Elements.ForEach(x => x.Accept(visitor));
        }
    }
}